test/lib/jdk/test/lib/crac/CracBuilder.java

```diff
@@ -331,7 +335,9 @@ private List<String> prepareCommand(List<String> javaPrefix, boolean isRestore)
         }
         cmd.add("-ea");
         if (engine != null) {
-            cmd.add("-XX:CREngine=" + engine.engine);
+            String engArgs = engineArgs == null ? "" : "," + Arrays.stream(engineArgs)
```

Comment:
I suppose `engineArgs == null` should be replaced with something like `engineArgs == null || 0 == engineArgs.length`

Reply:
You're right, I must have had something cached because when I re-ran the test it was passing... until I added some debug logs to confirm what you're seeing. I'll file a fix in a sec.